File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
# numbers 원소를 str 타입으로 변환 (이어붙일 수 있도록)
# numbers 원소는 1000 이하 => 따라서 모든 원소를 3배 (수를 3배 X, 문자열자체를 3배)

def solution(numbers):
    numbers = list(map(str, numbers))
    numbers.sort(key= lambda x:x*3, reverse=True)
    
    return str(int(''.join(numbers)))

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;
class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        
        String[] str = new String[numbers.length];
        for (int i=0; i<numbers.length; i++) {
            str[i] = String.valueOf(numbers[i]);
        }
        
        Arrays.sort(str, (o2, o1) -> (o1+o2).compareTo(o2+o1));
        
        // 첫숫자가 0인경우 그냥 0임
        if (str[0].equals("0")) {
            return "0";
        }
        
        for (int i=0; i<str.length; i++) {
            answer += str[i];
        }
        
        return answer;
    }
}